src/server/json_family.cc

```diff
@@ -1651,12 +1650,19 @@ void JsonFamily::Clear(CmdArgList args, ConnectionContext* cntx) {
 void JsonFamily::StrAppend(CmdArgList args, ConnectionContext* cntx) {
   string_view key = ArgS(args, 0);
   string_view path = ArgS(args, 1);
+  string_view value = ArgS(args, 2);
 
   WrappedJsonPath json_path = GET_OR_SEND_UNEXPECTED(ParseJsonPath(path));
-  auto strs = args.subspan(2);
 
+  // We try parsing the value into json string object first.
+  optional<JsonType> parsed_json = dfly::JsonFromString(value, PMR_NS::get_default_resource());
```

Comment:
Can we use [local `JsonFromString(value)`](https://github.com/dragonflydb/dragonfly/blob/main/src/server/json_family.cc#L253) here?

Reply:
no we can not, because we use shard allocators only on shard threads and it's not guaranteed that your calling thread is a shard thread.

Reply:
Yes, I'm asking because we are calling this function also from coordinator thread in another commands, e.g. [ArrIndex](https://github.com/dragonflydb/dragonfly/blob/main/src/server/json_family.cc#L1479)

Reply:
So, we need to fix it